### ISSUE TITLE
RuntimeError: Failed to read the record.

### ISSUE THREAD

**Issue by Vipermdl** [vahidk/tfrecord#2]
Hi, guys, thanks your contribution!

when I use your source code, I find this error:
Traceback (most recent call last):
  File "D:/coding/SIFA-master/model.py", line 316, in <module>
    data = next(iter(loader))
  File "D:\Anaconda3\lib\site-packages\torch\utils\data\dataloader.py", line 346, in __next__
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "D:\Anaconda3\lib\site-packages\torch\utils\data\_utils\fetch.py", line 28, in fetch
    data.append(next(self.dataset_iter))
  File "D:\Anaconda3\lib\site-packages\tfrecord\io_utils.py", line 106, in tfrecord_loader
    for record in record_iterator:
  File "D:\Anaconda3\lib\site-packages\tfrecord\io_utils.py", line 73, in tfrecord_iterator
    yield from read_records()
  File "D:\Anaconda3\lib\site-packages\tfrecord\io_utils.py", line 67, in read_records
    raise RuntimeError("Failed to read the record.")
RuntimeError: Failed to read the record.

There is my code:

from tfrecord.torch.dataset import TFRecordDataset
from torch.utils.data import DataLoader

tfrecord_path = "./data/training_A/image_01.tfrecords"

dataset = TFRecordDataset(tfrecord_path, None, {"dsize_dim0":"int", "dsize_dim1":"int", "dsize_dim2":"int",
                                                "lsize_dim0":"int", "lsize_dim0":"int", "lsize_dim0":"int",
                                                "data_vol":"str", "label_vol":"str",
                                                })

loader = DataLoader(dataset, batch_size=1)

data = next(iter(loader))

print(data)

so have you fix the error or help me to find mistake? Thanks!

**Comment by vahidk** [comment]
How did you generate the tfrecord? Also "str" is not supported. You can use "byte", "float", or "int" as type.

**Comment by faheuer** [comment]
Hi, I am having the same issue. My TfRecord file contains "str" files. Can your code still load the other data in the TfRecord or is it entirely not possible?

**Comment by vahidk** [comment]
Let me rephrase what I said. TensorFlow feature doesn't support string. It only supports bytes. To store strings you have to encode it into bytes and then decode it. If you see a parsing error it may be that your data is compressed. Compressed TFrecords are not currently supported. If you generated the data, just generated it again uncompressed. You can also try to figure out what compression was used and modify the code to decompress it before reading. I can't help without knowing how the data was generated.
